test_agent/rust/src/main.rs

```diff
@@ -44,19 +56,47 @@ fn connect_to_socket(addr: &str, port: u16) -> Result<TcpStream, Box<dyn std::er
     }
 }
 
-async fn connect_and_receive() -> Result<(), Box<dyn std::error::Error>> {
+async fn create_zenoh_u_transport() -> Box<dyn UTransport> {
+    let uauthority = UAuthority {
+        name: Some("MyAuthName".to_string()),
+        number: Some(Number::Id(vec![1, 2, 3, 4])),
+        ..Default::default()
+    };
+    let uentity = UEntity {
+        name: "default.entity".to_string(),
+        id: Some(u32::from(rand::random::<u16>())),
+        version_major: Some(1),
+        version_minor: None,
+        ..Default::default()
+    };
+    dbg!("zenoh transport created successfully");
+
+    Box::new(
+        UPClientZenoh::new(Config::default(), uauthority, uentity)
+            .await
+            .unwrap(),
+    )
+}
+
+async fn connect_and_receive(transport_name: &str) -> Result<(), Box<dyn std::error::Error>> {
     let test_agent = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
     let ta_to_tm_socket = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
     let foo_listener_socket_to_tm = connect_to_socket(TEST_MANAGER_ADDR.0, TEST_MANAGER_ADDR.1)?;
 
-    let u_transport = UTransportSocket::new()?;
-    debug!("Socket transport created successfully");
+    let u_transport: Box<dyn UTransport> = if transport_name == ZENOH_TRANSPORT {
```

Comment:
Try this:

(note that I also removed u_transport_ref since it's not needed)

```rust
    // we allow this because we'll have further transports we want to support and match works well
    // for that
    #[allow(clippy::single_match_else)]
    let u_transport: Box<dyn UTransport> = match transport_name {
        ZENOH_TRANSPORT => create_zenoh_u_transport().await,
        _ => {
            debug!("Socket transport created successfully");
            Box::new(UTransportSocket::new()?)
        }
    };

    let foo_listener = Arc::new(ListenerHandlers::new(foo_listener_socket_to_tm));
    let agent = SocketTestAgent::new(test_agent, foo_listener);
    agent
        .clone()
        .receive_from_tm(u_transport, ta_to_tm_socket)
        .await;
```

Reply:
Yep this works and the reason makes sense, integrated it :)